src/Spago/GlobalCache.hs

```diff
@@ -99,16 +99,20 @@ getMetadata cacheFlag = do
         Nothing -> mempty
         Just a  -> a
 
-      downloadMeta = do
-        metaBS <- (Http.httpBS metaURL >>= pure . Http.getResponseBody)
-        case decodeStrict' metaBS of
-          Nothing -> do
+      downloadMeta = handleAny
+        (const $ do
             echo "WARNING: Unable to download GitHub metadata, global cache will be disabled"
-            pure mempty
-          Just meta -> do
-            assertDirectory globalCacheDir
-            liftIO $ BS.writeFile globalPathToMeta metaBS
-            pure meta
+            pure mempty)
+        (do
+            metaBS <- Http.getResponseBody `fmap` Http.httpBS metaURL
```

Comment:
If you'd like to be able to customize the location of the metadata (because e.g. you have a proxy to GitHub in your network, or you could download it locally, etc) we could try to read it from environment, e.g. something like:

```suggestion
            metaBS <- Dhall.input Dhall.lazyText ("env:SPAGO_METADATA_JSON ? " <> metaURL <> " as Text")
```

This won't compile because types mismatch but just to give you an idea. This Dhall string:
- tries to read the `SPAGO_METADATA_JSON` variable from environment
- if that's missing, it tries to import the default URL `as Text`
- imports the result as a lazy `Text`, so then we can parse it

When setting the `SPAGO_METADATA_JSON` variable, you could put there something like `https://some.proxy/metadata.json as Text` or `./local/metadata.json as Text` or even `"{ \"some\": \"escaped json\" }"`

Reply:
I really like this idea. Will experiment with it a bit 👍